src/cpp/common/assembler_state.h

```diff
@@ -223,6 +226,19 @@ class assembler_state : public std::enable_shared_from_this<assembler_state>
     return m_dependent_labelmap;
   }
 
+  void process_optimization(uint32_t optimize_level)
+  {
+    if (optimize_level >= 1) {
```

Comment:
You can drop **{}** if you have only statement inside. That makes the code more succinct.